app/routes/group.$synonymId.tsx

```diff
@@ -0,0 +1,81 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import type { LoaderFunctionArgs } from '@remix-run/node'
+import { Link, useLoaderData } from '@remix-run/react'
+import { Icon } from '@trussworks/react-uswds'
+
+import type { SynonymGroup } from './synonyms/synonyms.lib'
+import {
+  getAllSynonymMembers,
+  getSynonymById,
+} from './synonyms/synonyms.server'
+import { ToolbarButtonGroup } from '~/components/ToolbarButtonGroup'
+import { PlainTextBody } from '~/components/circularDisplay/Body'
+import { FrontMatter } from '~/components/circularDisplay/FrontMatter'
+import { feature } from '~/lib/env.server'
+
+export async function loader({
+  params: { synonymId },
+  request: { url },
+}: LoaderFunctionArgs) {
+  if (!feature('SYNONYMS')) throw new Response(null, { status: 403 })
+  if (!synonymId) throw new Response('Id is required', { status: 400 })
+
+  const { searchParams } = new URL(url)
+  const view = searchParams.get('view') || 'group'
+  const limit = searchParams.get('limit') || 20
+  const page = searchParams.get('page') || 1
+  const synonym = (await getSynonymById(synonymId)) as SynonymGroup
+  const members = await getAllSynonymMembers(synonym.eventIds)
+
+  return { members, eventIds: synonym.eventIds, view, limit, page }
+}
+
+export default function Group() {
+  const { members, eventIds, view, limit, page } =
+    useLoaderData<typeof loader>()
+  const searchString = `?view=${view}&limit=${limit}&page=${page}`
```

Comment:
Can you use URLSearchParams or similar here to construct this query string in a more structured way, rather than doing manual string formatting?